modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/datatypes/utils/DifferentFamiliesPair.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.planner.datatypes.utils;
+
+import org.apache.ignite.internal.type.NativeType;
+import org.apache.ignite.internal.type.NativeTypes;
+
+/**
+ * Enumerates possible pairs of types belonging to a different type families for test purposes.
+ */
+public enum DifferentFamiliesPair implements TypePair {
```

Comment:
can we have test to check that we have pairs for all type families?

Reply:
good point, test has been added (`DifferentFamiliesPairTest`)